test/operator-framework-e2e/operator_framework_test.go

```diff
@@ -0,0 +1,977 @@
+	imageRef      string // Stores the bundle image reference
+}
+
+type CatalogDInfo struct {
+	baseFolderPath     string // Base path of the folder for the catalogs
+	catalogDir         string // The folder storing the FBC
+	operatorName       string // Name of the operator to be installed from the bundles
+	desiredChannelName string // Desired channel name for the operator
+	imageRef           string // Stores the FBC image reference
+	fbcFileName        string // Name of the FBC file
+}
+
+type OperatorActionInfo struct {
+	installVersion string // Version of the operator to be installed on the cluster
+	upgradeVersion string // Version of the operator to be upgraded on the cluster
+}
+
+type SdkProjectInfo struct {
+	projectName string
+	domainName  string
+	group       string
+	version     string
+	kind        string
+}
+
+const (
+	remoteRegistryRepo     = "localhost:5000/"
+	kindServer             = "operator-controller-op-dev-e2e"
+	deployedNameSpace      = "rukpak-system"
+	operatorControllerHome = "../.."
+)
+
+func TestOperatorFramework(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Operator Framework E2E Suite")
+}
+
+var _ = BeforeSuite(func() {
+
+	var err error
+	cfg = ctrl.GetConfigOrDie()
+
+	scheme := runtime.NewScheme()
+
+	err = catalogd.AddToScheme(scheme)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = operatorv1alpha1.AddToScheme(scheme)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = rukpakv1alpha1.AddToScheme(scheme)
+	Expect(err).ToNot(HaveOccurred())
+
+	c, err = client.New(cfg, client.Options{Scheme: scheme})
+	Expect(err).ToNot(HaveOccurred())
+
+	ctx = context.Background()
+
+})
+
+var _ = Describe("Operator Framework E2E for plain bundles", func() {
+	var (
+		bundleInfo      *BundleInfo
+		catalogDInfo    *CatalogDInfo
+		operatorAction  *OperatorActionInfo
+		operatorCatalog *catalogd.Catalog
+		operator        *operatorv1alpha1.Operator
+		err             error
+	)
+	BeforeEach(func() {
+		bundleInfo = &BundleInfo{
+			baseFolderPath: "../../testdata/bundles/plain-v0",
+			bundles: []BundleContent{
+				{
+					bInputDir:     "plain.v0.1.0",
+					bundleVersion: "0.1.0",
+				},
+				{
+					bInputDir:     "plain.v0.1.1",
+					bundleVersion: "0.1.1",
+				},
+			},
+		}
+		catalogDInfo = &CatalogDInfo{
+			baseFolderPath:     "../../testdata/catalogs",
+			fbcFileName:        "catalog.yaml",
+			operatorName:       "plain",
+			desiredChannelName: "beta",
+		}
+		operatorAction = &OperatorActionInfo{
+			installVersion: "0.1.0",
+			upgradeVersion: "0.1.1",
+		}
+		for i, b := range bundleInfo.bundles {
+			bundleInfo.bundles[i].imageRef = remoteRegistryRepo + catalogDInfo.operatorName + "-bundle:v" + b.bundleVersion
+		}
+		catalogDInfo.catalogDir = catalogDInfo.operatorName + "-catalog"
+		catalogDInfo.imageRef = remoteRegistryRepo + catalogDInfo.catalogDir + ":test"
+	})
+	When("Build and load plain+v0 bundle images into the test environment", func() {
```

Comment:
Rather than directly committing plain bundle manifests, it should be possible to generate the manifests from an SDK project with `kustomize build config/default > ${bundleRoot}/manifests/manifest.yaml`